worker/wallet.js

```diff
@@ -252,103 +222,20 @@ export async function checkWithdrawal ({ data: { hash, withdrawal, invoice }, bo
   // nothing to do if the withdrawl is already recorded and it isn't an invoiceForward
   if (!dbWdrwl) return
 
-  let wdrwl
-  let notSent = false
-  try {
-    wdrwl = withdrawal ?? await getPayment({ id: hash, lnd })
-  } catch (err) {
-    if (err[1] === 'SentPaymentNotFound' &&
-      dbWdrwl.createdAt < datePivot(new Date(), { milliseconds: -LND_PATHFINDING_TIMEOUT_MS * 2 })) {
-      // if the payment is older than 2x timeout, but not found in LND, we can assume it errored before lnd stored it
-      notSent = true
-    } else {
-      throw err
-    }
-  }
-
-  const logger = walletLogger({ models, wallet: dbWdrwl.wallet })
+  const wdrwl = withdrawal ?? await getPaymentOrNotSent({ id: hash, lnd, createdAt: dbWdrwl.createdAt })
 
   if (wdrwl?.is_confirmed) {
-    if (dbWdrwl.invoiceForward.length > 0) {
-      return await paidActionForwarded({ data: { invoiceId: dbWdrwl.invoiceForward[0].invoice.id, withdrawal: wdrwl, invoice }, models, lnd, boss })
+    if (dbWdrwl.invoiceForward) {
+      return await paidActionForwarded({ data: { invoiceId: dbWdrwl.invoiceForward.invoice.id, withdrawal: wdrwl, invoice }, models, lnd, boss })
     }
 
-    const fee = Number(wdrwl.payment.fee_mtokens)
-    const paid = Number(wdrwl.payment.mtokens) - fee
-    const [[{ confirm_withdrawl: code }]] = await serialize([
-      models.$queryRaw`SELECT confirm_withdrawl(${dbWdrwl.id}::INTEGER, ${paid}, ${fee})`,
-      models.withdrawl.update({
-        where: { id: dbWdrwl.id },
-        data: {
-          preimage: wdrwl.payment.secret
-        }
-      })
-    ], { models })
-    if (code === 0) {
-      notifyWithdrawal(dbWdrwl.userId, wdrwl)
-
-      const { request: bolt11, secret: preimage } = wdrwl.payment
-
-      logger?.ok(
-        `↙ payment received: ${formatSats(msatsToSats(paid))}`,
-        {
-          bolt11,
-          preimage,
-          fee: formatMsats(fee)
-        })
-    }
-  } else if (wdrwl?.is_failed || notSent) {
-    if (dbWdrwl.invoiceForward.length > 0) {
-      return await paidActionFailedForward({ data: { invoiceId: dbWdrwl.invoiceForward[0].invoice.id, withdrawal: wdrwl, invoice }, models, lnd, boss })
+    return await payingActionConfirmed({ data: { withdrawalId: dbWdrwl.id, withdrawal: wdrwl }, models, lnd, boss })
```

Comment:
The new paying action state machine doesn't require migration because it reuses the exists `status` field for transitions.